Sources/LiveViewNativeCharts/Modifiers/View Fundamentals Modifiers/ChartForegroundStyleScale.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  ChartForegroundStyleScale.swift
+//  
+//
+//  Created by murtza on 28/06/2023.
+//
+
+import Charts
+import SwiftUI
+import LiveViewNative
+
+/// The ``mapping`` Maps data categories to foreground styles.
+///
+/// See ``LiveViewNativeCharts/Charts/BasicChartSymbolShape`` for a list of possible values.
+///
+/// ```html
+/// <Chart modifiers={chart_foreground_style_scale(mapping: %{
+///   "A" => :circle,
+///   "B" => :asterisk,
+///   "C" => :plus
+/// })}>
+///   ...
+/// </Chart>
+/// ```
+///
+/// The possible data values plotted as foreground style in the chart. You can define the ``domain`` with an array for categorical values (e.g., ["A", "B", "C"])
+///
+/// See ``LiveViewNativeCharts/Charts/AnyScaleDomain`` for more details.
+///
+/// ```html
+/// <Chart modifiers={chart_foreground_style_scale(domain: ["A", "B", "C"])}>
+///   ...
+/// </Chart>
+/// ```
+///
+/// The ``range`` of foreground styles that correspond to the scale domain.
+///
+/// See ``LiveViewNativeCharts/Charts/BasicChartSymbolShape`` for a list of possible values.
+///
+/// ```html
+/// <Chart modifiers={chart_foreground_style_scale(range: [:circle, :asterisk, :plus])}>
+///   ...
+/// </Chart>
+/// ```
+///
+/// The ``type`` provides the ways you can scale the domain or range of a plot.
+///
+/// See ``LiveViewNativeCharts/Charts/ScaleType`` for a list of possible values.
+///
+/// ```html
+/// <Chart modifiers={chart_foreground_style_scale(type: :category)}>
+///   ...
+/// </Chart>
+/// ```
+///
+/// ## Arguments
+/// * ``mapping``
+/// * ``domain``
+/// * ``range``
+/// * ``type``
+#if swift(>=5.8)
+@_documentation(visibility: public)
+#endif
+struct ChartForegroundStyleScaleModifier: ViewModifier, Decodable {
+    /// The ``mapping`` Maps data categories to foreground styles.
+    ///
+    /// Create a mapping between a string and a ``LiveViewNativeCharts/Charts/BasicChartSymbolShape``.
+    ///
+    /// ```elixir
+    /// %{
+    ///     "A" => :circle,
+    ///     "B" => :asterisk,
+    ///     "C" => :square
+    /// }
+    /// ```
+    #if swift(>=5.8)
+    @_documentation(visibility: public)
+    #endif
+    private let mapping: [String:BasicChartSymbolShape]?
```

Comment:
This mapping should be from `String` to `AnyShapeStyle`, not a symbol shape. It will need to be updated throughout the modifier. The doc comments will need to be updated as well.